### PR TITLE
PLT-167 Fixes issue with trying to comment on a deleted post

### PR DESCRIPTION
**Pull request by rgarmsen2295**
Fixes a relatively rare issue that occurred when trying to comment on a post that had been deleted.  The previously used `post_deleted_modal` had stopped working consistently as well the pending post never being properly removed.

The modal should now appear consistently and the RHS should close along with the pending post being properly removed.  Also added in some additional checks for the RHS for rare instances where certain data no longer existed when dealing with deleted posts and the RHS.  A few loosely related issues remain and have been filed as bugs.

### CODE REVIEW

File: web/react/components/post_deleted_modal.jsx [hunk 1]
```diff
@@ -2,13 +2,43 @@
...
 
+        this.handleClose = this.handleClose.bind(this);
+
         this.state = {};
     }
+    componentDidMount() {
+        const self = this;
+
+        $(React.findDOMNode(this.refs.modal)).on('hidden.bs.modal', () => {
+            self.handleClose();
```
Comment (jwilander) [inline web/react/components/post_deleted_modal.jsx]: `this` gets automatically bound when using `=>` so no need for the 'self' declaration

Comment (rgarmsen2295) [inline web/react/components/post_deleted_modal.jsx]: That's super nice, thanks (no wonder we have a warning set to prefer it)!  I'll change it over later today when I get a chance.

**Comment by rgarmsen2295** [comment]
Ok removed the reference to self, ready for re-review @jwilander.

**Comment by jwilander** [comment]
+1

**Comment by hmhealey** [comment]
Looks good

**Pull request merged**